demos/mqtt/mqtt_demo_lightweight/mqtt_demo_lightweight.c

```diff
@@ -0,0 +1,1046 @@
+}
+/*-----------------------------------------------------------*/
+
+static void mqttPublishToTopic( int tcpSocket,
+                                MQTTFixedBuffer_t * pFixedBuffer )
+{
+    MQTTStatus_t result;
+    MQTTPublishInfo_t mqttPublishInfo;
+    size_t remainingLength;
+    size_t packetSize = 0;
+    size_t headerSize = 0;
+    uint8_t * pPacketIdentifierHigh;
+    int status;
+
+    /***
+     * For readability, error handling in this function is restricted to the use of
+     * asserts().
+     ***/
+
+    /* Some fields not used by this demo so start with everything at 0. */
+    memset( ( void * ) &mqttPublishInfo, 0x00, sizeof( mqttPublishInfo ) );
+
+    /* This demo uses QOS0 */
+    mqttPublishInfo.qos = MQTTQoS0;
+    mqttPublishInfo.retain = false;
+    mqttPublishInfo.pTopicName = MQTT_EXAMPLE_TOPIC;
+    mqttPublishInfo.topicNameLength = ( uint16_t ) strlen( MQTT_EXAMPLE_TOPIC );
+    mqttPublishInfo.pPayload = MQTT_EXAMPLE_MESSAGE;
+    mqttPublishInfo.payloadLength = strlen( MQTT_EXAMPLE_MESSAGE );
+
+    /* Find out length of Publish packet size. */
+    result = MQTT_GetPublishPacketSize( &mqttPublishInfo, &remainingLength, &packetSize );
+    assert( result == MQTTSuccess );
+
+    /* Make sure the packet size is less than static buffer size. */
+    assert( packetSize < pFixedBuffer->size );
+
+    /* Serialize MQTT Publish packet header. The publish message payload will
+     * be sent directly in order to avoid copying it into the buffer.
+     * QOS0 does not make use of packet identifier, therefore value of 0 is used */
+    result = MQTT_SerializePublishHeader( &mqttPublishInfo,
+                                          0,
+                                          remainingLength,
+                                          pFixedBuffer,
+                                          &headerSize );
+    LogDebug( ( "Serialized PUBLISH header size is %lu.",
+                headerSize ) );
+    assert( result == MQTTSuccess );
+    /* Send Publish header to the broker. */
+    status = send( tcpSocket, ( void * ) pFixedBuffer->pBuffer, headerSize, 0 );
+    assert( status == ( int ) headerSize );
+    /* Send Publish payload to the broker */
+    status = send( tcpSocket, ( void * ) mqttPublishInfo.pPayload, mqttPublishInfo.payloadLength, 0 );
+    assert( status == ( int ) mqttPublishInfo.payloadLength );
+}
+/*-----------------------------------------------------------*/
+
+static void mqttUnsubscribeFromTopic( int tcpSocket,
+                                      MQTTFixedBuffer_t * pFixedBuffer )
+{
+    MQTTStatus_t result;
+    MQTTSubscribeInfo_t mqttSubscription[ 1 ];
+    size_t remainingLength;
+    size_t packetSize;
+    int status;
+
+    /* Some fields not used by this demo so start with everything at 0. */
+    memset( ( void * ) &mqttSubscription, 0x00, sizeof( mqttSubscription ) );
+
+    /* Unsubscribe to the MQTT_EXAMPLE_TOPIC topic filter. The task handle is passed
```

Comment:
Which task handle?

Reply:
Does not make sense at all. Thanks for reading all comments in detail.